RESTResource/RESTResource.js

```diff
@@ -256,6 +256,39 @@ export function substitute(original, props, state, module, logger, dataKey) {
   return result;
 }
 
+/**
+ * buildOption
+ * Some manifest options properties may be an object or a function, determine which
+ * the passed property is and act accordingly
+ *
+ * @param option object or function
+ * @param props object
+ * @param state object
+ * @param module string name of the module the resource is affiliated with
+ * @param logger object a logger
+ * @param dataKey string unique key to disambiguate this resource from another
+ * which otherwise has the same attributes
+ *
+ * @return object
+ */
+
+export function buildOption(option, props, state, module, logger, dataKey) {
+  let toReturn;
+  if (typeof option === 'object') {
+    toReturn = _.mapValues(
+      option,
+      param => substitute(param, props, state, module, logger, dataKey)
+    );
+    // If any of the option properties have null values, we can't go any further
+    const someParamIsNull = Object.values(toReturn).some(value => value === null);
+    if (someParamIsNull) return null;
```

Comment:
I would aim for consistency here. Since you have `toReturn`; use it.

Reply:
Good point, thanks for highlighting it.